### ISSUE TITLE
So Mac can not use this?

### ISSUE THREAD

**Issue by iwoomi** [microsoft/JARVIS#39]
Macs are not using NVIDIA display card, so Mac can not use this right?

**Comment by xiebruce** [comment]
> I guess you still can, but using `hybrid` mode only. https://github.com/microsoft/JARVIS#configuration

But the server needs Nvidia display card.
<img width="842" alt="image" src="https://user-images.githubusercontent.com/13575500/230059002-d5f16e63-998f-460a-9c89-159b925a6551.png">

<img width="620" alt="image" src="https://user-images.githubusercontent.com/13575500/230059299-1a2a77c6-4276-44bf-8646-4c8195f5704c.png">

**Comment by Fermain** [comment]
There are various ways to configure this package depending on your resource limitations. I am using it on a Mac right now:

<img width="360" alt="image" src="https://user-images.githubusercontent.com/10801540/230134492-d7873336-0d40-437d-8ecb-ab3e7bf2144d.png">

**Comment by ethanye77** [comment]
> There are various ways to configure this package depending on your resource limitations. I am using it on a Mac right now:
> 
> <img alt="image" width="360" src="https://user-images.githubusercontent.com/10801540/230134492-d7873336-0d40-437d-8ecb-ab3e7bf2144d.png">

![WX25660405-090513@2x](https://user-images.githubusercontent.com/128858378/230138982-84f9fc97-c4a7-4472-be0e-7d7718201cae.png)
I am also a Mac user and I encountered this issue while running this line of code. Could you please tell me what I should do if it is convenient？

**Comment by ethanye77** [comment]
> 
here‘s my issue
<img width="605" alt="image" src="https://user-images.githubusercontent.com/128858378/230139840-aa8bb05a-0645-4bac-a902-56a98c8bd9a6.png">

**Comment by ErikDombi** [comment]
Yes, the LFS objects are rather large. My models folder is 275 GB personally.

**Comment by davidjhwu** [comment]
Are the LFS objects absolutely necessary?
Tryna run this on my macbook air lol (16gb ram, 500gb ssd)

**Comment by Fermain** [comment]
> Are the LFS objects absolutely necessary? Tryna run this on my macbook air lol (16gb ram, 500gb ssd)

No, you can run the `lite.yaml` configuration to use remote models only, although this is quite limited at the moment. I suggest using an external hard drive or SSD to manage these large models.

**Comment by iwoomi** [comment]
@Fermain So If we deploy JARVIS in macOS, we can only use the `lite.yaml`(that is `inference_mode: huggingface`)  right? Because if we use `inference_mode:local`(or `inference_mode:hybrid`)，we should have a Nvidia display card, but Macs have no Nvidia display card, is that right?

**Comment by sirlaurie** [comment]
> @Fermain So If we deploy JARVIS in macOS, we can only use the `lite.yaml`(that is `inference_mode: huggingface`) right? Because if we use `inference_mode:local`(or `inference_mode:hybrid`)，we should have a Nvidia display card, but Macs have no Nvidia display card, is that right?

comment line 297-299(maybe if you didn't reformat this file) in models_server.py file

> "midas-control": {sometmodel here} 

you can run without nvidia device.

**Comment by van0303** [comment]
I have just downloaded the models on my Mac, I don't have the N display card.
And i have started with this ```models_server.py --config lite.yaml```
I got the error messages :
`AssertionError: Torch not compiled with CUDA enabled` 

comment the 
```
"midas-control": {
            "model": MidasDetector(model_path=f"{local_fold}/lllyasviel/ControlNet/annotator/ckpts/dpt_hybrid-midas-501f0c75.pt")
            }
```

the models_server started

**Comment by xmagicwu** [comment]
<img width="1289" alt="image" src="https://user-images.githubusercontent.com/1046268/230538263-6812b66d-358b-4131-b7bf-673c2e70b1b7.png">
Encountered a problem when executing here, what should I do here?

**Comment by sirlaurie** [comment]
did you run `git lfs install`?

**Comment by xmagicwu** [comment]
yes,git lfs installed

**Comment by xmagicwu** [comment]
the version is 3.3.0

**Comment by sirlaurie** [comment]
I mean after you installed git-lfs, you need run `git lfs install` first

if you did it already, run ` sh download.sh` again

**Comment by xmagicwu** [comment]
Thanks, I'll try it

**Comment by xiebruce** [comment]
> There are various ways to configure this package depending on your resource limitations. I am using it on a Mac right now:
> 
> <img alt="image" width="360" src="https://user-images.githubusercontent.com/10801540/230134492-d7873336-0d40-437d-8ecb-ab3e7bf2144d.png">

did you encounter this error:  https://github.com/microsoft/JARVIS/issues/67

> Solving environment: failed with initial frozen solve.

**Comment by liujie316316** [comment]
> Thanks, I'll try it

Hello, have you resolved this issue? I also reported the same error.
![image](https://user-images.githubusercontent.com/44899079/230711337-817917dc-14a0-4075-a31c-4b48d12cf2bd.png)

I executed the following command but still reported an error:
pip install git-lfs
cd models
sh download.sh

**Comment by Fermain** [comment]
> I executed the following command but still reported an error: pip install git-lfs cd models sh download.sh

git-lfs is not a pip package. You can use homebrew to install it:

```sh
brew install git-lfs
```

The error message states that this is not installed.

**Comment by liujie316316** [comment]
> > I executed the following command but still reported an error: pip install git-lfs cd models sh download.sh
> 
> git-lfs is not a pip package. You can use homebrew to install it:
> 
> ```shell
> brew install git-lfs
> ```
> 
> The error message states that this is not installed.

OK,Thank you!

**Comment by xmagicwu** [comment]
<img width="1208" alt="image" src="https://user-images.githubusercontent.com/1046268/230846557-da367b8c-4b61-47fc-810c-53dda409fb78.png">
my device is mackbook M1,

**Comment by xmagicwu** [comment]
<img width="1206" alt="image" src="https://user-images.githubusercontent.com/1046268/230847126-9647ce88-ce88-4a1f-8310-b3bb1a36dad7.png">
My device is a mackbook M1, how to solve this problem？

**Comment by Fermain** [comment]
Without Nvidia hardware, there is no solution to this particular issue. This system is not designed to run on Apple hardware and can only be used in limited ways on this platform.

**Comment by xmagicwu** [comment]
How to use it restrictively?

**Comment by Fermain** [comment]
The readme contains instructions for using the model with the `lite.yaml` config file instead of the full `config.yaml` file. Add your API keys to this lite file, and run this instead of `config`.

**Comment by sirlaurie** [comment]
> <img alt="image" width="1206" src="https://user-images.githubusercontent.com/1046268/230847126-9647ce88-ce88-4a1f-8310-b3bb1a36dad7.png">
> My device is a mackbook M1, how to solve this problem？

checkout my first post in this issue:

https://github.com/microsoft/JARVIS/issues/39#issuecomment-1499319851

**Comment by xmagicwu** [comment]
python models_server.py --config lite.yaml

**Comment by xmagicwu** [comment]
> > <img alt="图像" width="1206" src="https://user-images.githubusercontent.com/1046268/230847126-9647ce88-ce88-4a1f-8310-b3bb1a36dad7.png">
> > 我的设备是mackbook M1，如何解决这个问题？
> 
> 查看我在本期中的第一篇文章：
> 
> [#39（评论）](https://github.com/microsoft/JARVIS/issues/39#issuecomment-1499319851)
> 
> 你不需要`config.yaml`改成`lite.yaml`

<img width="1027" alt="image" src="https://user-images.githubusercontent.com/1046268/230867373-64ad03f0-a7b7-4a82-89f0-888b57cd37f7.png">
Did it work successfully?

**Comment by sirlaurie** [comment]
it did

**Comment by Fermain** [comment]
@sirlaurie I missed that comment, very helpful - thanks

**Comment by iwoomi** [comment]
> Without Nvidia hardware, there is no solution to this particular issue. This system is not designed to run on Apple hardware and can only be used in limited ways on this platform.

@sirlaurie @Fermain I notice that we can config the device to "cuda" or "cpu" in [here](https://github.com/microsoft/JARVIS/blob/main/server/config.yaml#L12)
```
device: cuda:0 # cuda:id or cpu
```

Do it mean that if I set the device to "cpu", then I can run it on Mac, no matter M1/M2 chip(new Mac) or Intel cpu(old Mac) ?

**Comment by xmagicwu** [comment]
> > @Fermain So If we deploy JARVIS in macOS, we can only use the `lite.yaml`(that is `inference_mode: huggingface`) right? Because if we use `inference_mode:local`(or `inference_mode:hybrid`)，we should have a Nvidia display card, but Macs have no Nvidia display card, is that right?
> 
> comment line 298-300(maybe if you didn't reformat this file) in models_server.py file
> 
> `"midas-control": {sometmodel here}`
> 
> you can run without nvidia device.

very helpful - thanks

But encountered another problem～
<img width="844" alt="image" src="https://user-images.githubusercontent.com/1046268/230892063-07044ca0-d6e3-4fe2-86e1-0ad81124142c.png">

**Comment by sirlaurie** [comment]
> 

looks like it's a newly added option, but unfortunately, still no

**Comment by sirlaurie** [comment]
> > Without Nvidia hardware, there is no solution to this particular issue. This system is not designed to run on Apple hardware and can only be used in limited ways on this platform.
> 
> @sirlaurie @Fermain I notice that we can config the device to "cuda" or "cpu" in [here](https://github.com/microsoft/JARVIS/blob/main/server/config.yaml#L12)
> 
> ```
> device: cuda:0 # cuda:id or cpu
> ```
> 
> Do it mean that if I set the device to "cpu", then I can run the server on `inference_mode =local` on Mac, no matter M1/M2 chip(new Mac) or Intel cpu(old Mac) ?

looks like it's a newly added option, but unfortunately, still no

**Comment by sirlaurie** [comment]
> > > @Fermain So If we deploy JARVIS in macOS, we can only use the `lite.yaml`(that is `inference_mode: huggingface`) right? Because if we use `inference_mode:local`(or `inference_mode:hybrid`)，we should have a Nvidia display card, but Macs have no Nvidia display card, is that right?
> > 
> > 
> > comment line 298-300(maybe if you didn't reformat this file) in models_server.py file
> > `"midas-control": {sometmodel here}`
> > you can run without nvidia device.
> 
> very helpful - thanks
> 
> But encountered another problem～ My hugginggpt not work~ <img alt="image" width="844" src="https://user-images.githubusercontent.com/1046268/230892063-07044ca0-d6e3-4fe2-86e1-0ad81124142c.png">

check your network or your api quota

**Comment by xmagicwu** [comment]
> 

thanks

<img width="380" alt="image" src="https://user-images.githubusercontent.com/1046268/231144163-dc6705b3-f48c-4405-9a68-6926da34b882.png">
How can the generated pictures be accessed?

**Comment by xiebruce** [comment]
> > 
> 
> thanks
> 
> <img alt="image" width="380" src="https://user-images.githubusercontent.com/1046268/231144163-dc6705b3-f48c-4405-9a68-6926da34b882.png">
> 
> How can the generated pictures be accessed?

This is a bug, you should create "images" and "audios" folder under `/path/to/JARVIS/server/public/`, theoretically, the program should create these two folder automatically, but it didn't ,so this is a bug!

**Comment by xmagicwu** [comment]
<img width="445" alt="image" src="https://user-images.githubusercontent.com/1046268/231155189-187d1430-4cd4-4379-bf99-c3a1ba04bbb1.png">
<img width="559" alt="image" src="https://user-images.githubusercontent.com/1046268/231155252-fad33da3-be2f-42de-bcc0-61f7defdbfb4.png">
Why does the path for generating images keep changing?

**Comment by xmagicwu** [comment]
<img width="445" alt="image" src="https://user-images.githubusercontent.com/1046268/231160719-b3b03b15-e596-4c9c-b762-75efb77f4211.png">
what's wrong?

**Comment by sirlaurie** [comment]
I think the latest commit has fixed this bug. just pull again
